gadopt/level_set_tools.py

```
@@ -53,36 +53,46 @@
 ) -> fd.Function:
     """Default strategy for the thickness of the conservative-level-set profile.
 
     The interface thickness must not exceed the local minimum edge length to ensure
     materials remain immiscible. In practice, there exists a tradeoff between interface
     thickness and solution accuracy, for example, in terms of interface location and
-    material conservation. The default strategy implemented here is to multiply the
-    local minimum edge length by 0.35, but this choice can be modified via the `scale`
-    argument.
-
-    **Note**: UFL's `MinCellEdgeLength` is used to recover locally the minimum edge
-    length. This class, however, is not compatible with extruded meshes. In that case,
-    the `cell_sizes` attribute is used as an approximation, and the scaling factor
-    is divided by the square root of the mesh's geometric dimension. Do note that this
-    formulation is not equivalent and will result in measurable changes between
-    comparable extruded and non-extruded meshes.
+    material conservation.
+
+    UFL's `MinCellEdgeLength` can be used to recover locally the minimum edge length.
+    This class, however, is not compatible with extruded meshes or meshes for which the
+    polynomial degree of the coordinate function space exceeds 1. As a result, the
+    default strategy implemented here is to approximate the local minimum edge length
+    as the quotient of the `cell_sizes` mesh attribute and the square root of the mesh's
+    geometric dimension. If `MinCellEdgeLength` is desired instead, a value of `True`
+    must be provided for the `min_cell_edge_length` argument. The local minimum edge
+    length is then multiplied by a scaling factor, with a default value of 0.35 that can
+    be modified via the `scale` argument.
+
+    **Note**: The two different formulations implemented here are not equivalent and
+    will result in measurable changes, for example, between comparable extruded and
+    non-extruded meshes.
```

This last comment is fine, but should you also add a brief comment stating that cell_sizes yields the same behavior on both extruded and non-extruded meshes, for clarity?

---

Also if `MinCellEdgeLength` does not work on `ExtrudedMesh`, one should raise an error if the mesh is extruded.